omod-1.9/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs1_9/PatientIdentifierController1_9Test.java

```diff
@@ -189,4 +191,37 @@ public void shouldReturnTheAuditInfoForTheFullRepresentation() throws Exception
 		
 		assertNotNull(PropertyUtils.getProperty(result, "auditInfo"));
 	}
+
+	@Test
+	public void shouldNotAddIdentifierInUseByAnotherPatient() throws Exception {
+		SimpleObject patientIdentifier = new SimpleObject();
+		patientIdentifier.add("identifier", "123456789qwerty");
+		patientIdentifier.add("identifierType", "2f470aa8-1d73-43b7-81b5-01f0c0dfa53c");
+		patientIdentifier.add("location", RestTestConstants1_8.LOCATION_UUID);
+
+		String json = new ObjectMapper().writeValueAsString(patientIdentifier);
+
+		MockHttpServletRequest req = request(RequestMethod.POST, getURI());
+		req.setContent(json.getBytes());
+
+		SimpleObject newPatientIdentifier = deserialize(handle(req));
+
+		assertNotNull(PropertyUtils.getProperty(newPatientIdentifier, "uuid"));
+
+		final String OTHER_PATIENT_UUID = "5946f880-b197-400b-9caa-a3c661d23041";
+		final String REQUEST_URI = "patient/" + OTHER_PATIENT_UUID + "/identifier";
+
+		int otherPatientActiveIdentifiersSize = service.getPatientByUuid(OTHER_PATIENT_UUID).getActiveIdentifiers()
+				.size();
+		req = request(RequestMethod.POST, REQUEST_URI);
+		req.setContent(json.getBytes());
+		try {
+			handle(req);
+			fail();
```

Comment:
Do you mind explaining why you are using the above `fail()`?

Reply:
The idea is, If we add an identifier with a value that is already used by another patient, it should throw a validation exception. Should it not throw an exception, it means the save was a success and therefore this test should fail.

Reply:
Shouldn't the trowing of an exception happen automatically without having `fail()`?

Reply:
What I mean is, if `handle(req);` doesn't throw an exception, it means the `req` was successful, which it shouldn't be. 
If an exception occurs, execution won't get to `fail();` which marks this test as failed. 
The `fail();` should actually be more descriptive maybe `fail("This test fails here because the exception was not thrown yet it was expected");`

Reply:
Wouldn't it be clearer to just use ExpectedException

Reply:
Yes - except that in this case, After the exception is thrown, we need to check that the identifier wasn't persisted (which is the issue we were facing, that despite an exception being thrown, an identifier was actually persisted) 
@Test(expected = APIException) would pass once the exception is thrown and we wouldn't get to a point where we  test whether an identifier was persisted after the exception is thrown (like in the issue reported) or not!